### PR TITLE
DEVEXP-488: bump k8s 1.17.0-rc.1

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @adambkaplan 

@openshift/openshift-team-developer-experience fyi

similar approach to other 1.17 rebase PRs

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
+/workspace.xml
```
Comment (adambkaplan) [inline .idea/.gitignore]: Please update `.gitignore` to ignore `.idea/`

File: go.mod [hunk 2]
```diff
@@ -48,48 +46,45 @@ require (
...
+	k8s.io/component-base => k8s.io/component-base v0.0.0-20191122220729-2684fb322cb9
+	k8s.io/cri-api => k8s.io/cri-api v0.0.0-20191121183020-775aa3c1cf73
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.0.0-20191122224157-518651485193
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20191122221605-1e8d331e4dcc
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20191122223648-5cfd5067047c
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20191122223145-16f2c0c680a0
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191122223507-ac656df73673
+	k8s.io/kubectl => k8s.io/kubectl v0.0.0-20191122225023-1e3c8b70f494
+	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20191122223325-9316382755ad
+	k8s.io/kubernetes => k8s.io/kubernetes v1.17.0-rc.2
```
Comment (adambkaplan) [inline go.mod]: Is `-rc.2` not a valid semver for go.mod? Otherwise I would expect `k8s.io/kubernetes` to not need a replace stanza.

**Comment by gabemontero** [comment]
On Wed, Dec 4, 2019 at 11:19 AM Adam Kaplan <notifications@github.com>
wrote:

> *@adambkaplan* requested changes on this pull request.
> ------------------------------
>
> In .idea/.gitignore
> <https://github.com/openshift/builder/pull/118#discussion_r353840076>:
>
> > @@ -0,0 +1,2 @@
> +# Default ignored files
> +/workspace.xml
>
> Please update .gitignore to ignore .idea/
> ------------------------------
>
> In go.mod
> <https://github.com/openshift/builder/pull/118#discussion_r353841318>:
>
> > +	k8s.io/apiserver => k8s.io/apiserver v0.0.0-20191122221311-9d521947b1e1
> +	k8s.io/cli-runtime => k8s.io/cli-runtime v0.0.0-20191122222818-9150eb3ded31
> +	k8s.io/client-go => k8s.io/client-go v0.0.0-20191122220542-ed16ecbdf3a0
> +	k8s.io/cloud-provider => k8s.io/cloud-provider v0.0.0-20191122224014-59d76ad41876
> +	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.0.0-20191122223827-289de4a64c1c
> +	k8s.io/code-generator => k8s.io/code-generator v0.0.0-20191121175249-e95606b614f0
> +	k8s.io/component-base => k8s.io/component-base v0.0.0-20191122220729-2684fb322cb9
> +	k8s.io/cri-api => k8s.io/cri-api v0.0.0-20191121183020-775aa3c1cf73
> +	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.0.0-20191122224157-518651485193
> +	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20191122221605-1e8d331e4dcc
> +	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20191122223648-5cfd5067047c
> +	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20191122223145-16f2c0c680a0
> +	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191122223507-ac656df73673
> +	k8s.io/kubectl => k8s.io/kubectl v0.0.0-20191122225023-1e3c8b70f494
> +	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20191122223325-9316382755ad
> +	k8s.io/kubernetes => k8s.io/kubernetes v1.17.0-rc.2
>
> Is -rc.2 not a valid semver for go.mod? Otherwise I would expect
> k8s.io/kubernetes to not need a replace stanza.
>
I got errors when I tried it.

File: go.mod (hunk 2 above)
Comment (gabemontero) [reply go.mod]: I got errors when I tried it.

File: .idea/.gitignore (hunk 1 above)
Comment (gabemontero) [reply .idea/.gitignore]: dammit I thought I had handled that ... will update shortly

Comment (gabemontero) [reply .idea/.gitignore]: oh it was in the ocm PR :-)

Comment (gabemontero) [reply .idea/.gitignore]: ok now in builder PR as well

**Comment by adambkaplan** [comment]
@gabemontero squash commits, then with a few runs of aws-builds I think we can call this good.

/approve

**Comment by gabemontero** [comment]
commits squashed

**Comment by gabemontero** [comment]
all green tests for the builder 1.17 rebase @adambkaplan plus 2 clean e2e-aws-builds runs

/test e2e-aws-builds

**Comment by gabemontero** [comment]
initiated a 3rd e2e-aws-builds

**Comment by gabemontero** [comment]
another passing e2e-aws-builds

/test e2e-aws-builds

**Comment by gabemontero** [comment]
another passing e2e-aws-builds

I'm declaring victory for the builder 1.17 rebase 

@adambkaplan can you lgtm ?

**Pull request merged**